cmake/list_insert_options.cmake

```diff
@@ -0,0 +1,55 @@
+#
+# Append key[=value] elements from one list to another list if they are not already in the target list.
```

Comment:
> Depending on which CMake versions you need to support CMAKE_CXX_STANDARD might be an interesting option too (to be used in your enable_cxx11_support package).

The package was meant for `indigo` as well, so I don't want to rely on new CMake features.
It seems like most users have settled to set the C++11 options in any depending package :-/